src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
```

Comment:
@ahsonkhan, this whole file points at holes in our APIs. It would be great to see if we should not pull some of these into the BCL.

Reply:
Agreed. Most of them are either already scheduled for 3.0 or redundant.
- [Split](https://github.com/dotnet/corefx/issues/26528), we should probably add one for Memory\<T\> too.
- [TrimSpaces](https://github.com/dotnet/corefx/issues/31011)
- SplitOne [not sure about this one, although it is relatively common in practice]

The ones related to StringBuilder are interesting:
- AddToStringBuilder
- StringBuilder.Append

Reply:
@ahsonkhan, do we have issues for the stringbuilder and splitone operations?

Reply:
No, we don't, yet. I will create them soon.